### PR TITLE
Django mongo things

### PR DESCRIPTION
**Pull request by skosman**
Hey! Here are all the implemented user commands. I've also implemented a quote and mock quote server. 

A few things to note:
- I've never really done backend before especially with mongodb and databases in fullstack applications so please let me know what to fix!
- When running the current workload generator it uses the mock quote server that has been created.
- All commands work with WL user 1 and you should see reflections of the commands in the database. 
- Over reading break, I plan to extract each command into its own class to provide a more clarity and organization, as well as add more error catching (try/except)
- Last thing, I still haven't implement the constant checking of the quote server for the sell/buy triggers. The triggers get set in the db but aren't checked by the quote server. However I believe this shouldn't impact the xsd files.

### CODE REVIEW

File: backend/src/web_server.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ def start(self):
             self.socket.bind((self.hostname, self.port))
 
         except socket.error as err:
-            print("ERROR! binding port {self.port} failed with error: {err}")
+            print(f"ERROR! binding port {self.port} failed with error: {err}")
+            self.socket.close()
```
Comment (DamonAknuh) [inline backend/src/web_server.py:61]: I think shutdown before close is the order of operations.

File: backend/src/quoteServer/quoteServer.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	def getQuote(self, cmdDict):
+		"""
+			Get a quote from the quote server
+		"""
+
+		# Get query and send to quote server
+		quote =  cmdDict['stockSymbol'] + ',' + cmdDict['user'] + "\n"
+			
+		self.socket.send(quote.encode())
```
Comment (DamonAknuh) [inline backend/src/quoteServer/quoteServer.py:45]: I think from the comments in the tutorial we need to reinitialize the socket each time we have a query request.

File: backend/src/user_commands.py [hunk 3]
```diff
@@ -2,165 +2,517 @@
...
+    
 def CMD_SetBuyTrigger(cmdDict, threadContext):
+    """
+        Adds the price trigger to the set buy command
+    """
+
     printCmd(cmdDict)
 
+    # Check if person has already set buy amount for the stock
+    if  (Database.find_one(TRIGGER_COLLECT, {'user': cmdDict['user'], 'stockSymbol': cmdDict['stockSymbol'], 'type': 'buy'}) != None):
```
Comment (DamonAknuh) [inline backend/src/user_commands.py:400]: maybe a separate transaction server can handle these triggers aswell as query the stock server every 10ms?

**Review by DamonAknuh — APPROVED** [review]

File: backend/src/web_server.py (hunk 1 above)
Comment (skosman) [reply backend/src/web_server.py:61]: Yes you are correct! Do you want me to add it to the shutdown function instead so it happens every time shutdown is called?

File: backend/src/quoteServer/quoteServer.py (hunk 2 above)
Comment (skosman) [reply backend/src/quoteServer/quoteServer.py:45]: Yup, I'll do that.

File: backend/src/user_commands.py (hunk 3 above)
Comment (skosman) [reply backend/src/user_commands.py:400]: I think that's what Dianna and I were talking about on Thursday. It's a great idea, will do!

**Comment by DiannaBanana** [comment]
Looks great! I think the only thing were the logging bugs (I think we quote the quoteserver for the BUY & SELL commands, so that should be logged too)

File: backend/src/user_commands.py [hunk 4]
```diff
@@ -2,165 +2,523 @@
...
+            'stockSymbol': cmdDict['stockSymbol'],
+            'amount': cmdDict['amount']
+        }
+        Database.update_one(ACCOUNTS_COLLECT, {'_id': cmdDict['user']}, {'$set': { 'buy': stock_data}})
 
-    threadContext["buyAmtQ"].put(cmdDict['amount'])
     cmdDict['timestamp'] = str(int(time.time()))
+    cmdDict['amount'] = str(int(cmdDict['amount']))
+
     log.logEvents['userCommand'](cmdDict)
```
Comment (DiannaBanana) [inline backend/src/user_commands.py:145]: log quoteServer hit here

File: backend/src/user_commands.py [hunk 5]
```diff
@@ -2,165 +2,523 @@
...
+            'stockSymbol': cmdDict['stockSymbol'],
+            'amount': cmdDict['amount']
+        }
+        Database.update_one(ACCOUNTS_COLLECT, {'_id': cmdDict['user']}, {'$set': { 'sell': stock_data}})
 
-    threadContext["sellAmtQ"].put(cmdDict['amount'])
     cmdDict['timestamp'] = str(int(time.time()))
+    cmdDict['amount'] = str(int(cmdDict['amount']))
 
     log.logEvents['userCommand'](cmdDict)
```
Comment (DiannaBanana) [inline backend/src/user_commands.py:270]: log quoteServer hit here too

**Review by DiannaBanana — APPROVED** [review]

**Pull request merged**